### PR TITLE
update chain-id format

### LINKED ISSUE #540: Allow ChainId as string
__Proposal:__ ChainId should accept a string

__Current behavior:__ ChainId allows only base-10 numbers

__Desired behavior:__ ChainId accepts a string

__Use case:__ The current standard used by all known CosmosSDK blockchains are string as ChainId. E.g. cosmoshub-3

Even though the standard for Ethereum blockchains is using a numeric ChainId, in the Cosmos Ecosystem strings are mostly used and expected to be used in the future. Supporting string ChainId's would make the integration seamless for many projects.

Requests may be closed if we're not actively planning to work on them.

### PR DESCRIPTION
**Pull request by fedekunze**
closes #540

### CODE REVIEW

File: x/evm/handler.go [hunk 1]
```diff
@@ -30,13 +28,13 @@ func NewHandler(k Keeper) sdk.Handler {
 // handleMsgEthereumTx handles an Ethereum specific tx
 func handleMsgEthereumTx(ctx sdk.Context, k Keeper, msg types.MsgEthereumTx) (*sdk.Result, error) {
 	// parse the chainID from a string to a base-10 integer
-	intChainID, ok := new(big.Int).SetString(ctx.ChainID(), 10)
-	if !ok {
-		return nil, sdkerrors.Wrap(emint.ErrInvalidChainID, ctx.ChainID())
+	chainIDEpoch, err := ethermint.ParseChainID(ctx.ChainID())
+	if err != nil {
+		return nil, err
```
Comment (fedekunze) [inline x/evm/handler.go:33]: @noot this is where we parse the chain ID epoch number for ethereum compatibility

**Review by noot — APPROVED** [review]

**Review by noot — CHANGES_REQUESTED** [review]
getting some errors when sending a tx:
```
$ geth attach http://172.17.0.2:8545

> eth.sendTransaction({from: eth.accounts[0], to: eth.accounts[0], value: 0x1})
Error: invalid chain ID: 17: failed to simulate tx
	at web3.js:6347:37(47)
	at web3.js:5081:62(37)
	at <eval>:1:20(16)

```

**Comment by noot** [comment]
@fedekunze figured out the issue, could you change this so we cannot use an invalid chain ID (ie `--chain-id=17`) with  `ethermintd testnet` or `ethermintcli rest-server`?

**Review by noot — APPROVED** [review]

**Comment by fedekunze** [comment]
replaced by #542

**Pull request closed without merging**